lib/broccoli/ember-app.js

```diff
@@ -1024,13 +1024,26 @@ class EmberApp {
         overwrite: true,
       });
 
-      if (this.amdModuleNames) {
-        let anonymousAmd = new Funnel(externalTree, {
-          files: Object.keys(this.amdModuleNames),
-          annotation: 'Funnel (named AMD)',
-        });
-        externalTree = this._mergeTrees([externalTree, shimAmd(anonymousAmd, this.amdModuleNames)], {
-          annotation: 'TreeMerger (named AMD)',
+      if (this.entries) {
+        let transformedTrees = [];
+        for (let filePath in this.entries) {
+          let entry = this.entries[filePath];
+          let treeToBeTransformed = new Funnel(externalTree, {
```

Comment:
I agree with you. The primary reason I went with this approach is to respect the ordering. Example:

An addon imported a file and ran a bunch of transforms: a, b, and c. I thought we should make sure to run those in order. 

If ordering is not of importance and we run the AMD transform always last, then I can group the funnel operation to be per transform (which would do 1 funnel for all amd transforms) rather than per file.

@stefanpenner thoughts?